tests/integration/test_tiling.py

```diff
@@ -84,6 +94,32 @@ def test_tile_transform(self):
         num_tile_cols = (width + w_stride - 1) // w_stride
         assert len(tiled_dataset) == (num_tile_rows * num_tile_cols * len(dataset)), "Incorrect number of tiles"
 
+    def test_tiler_consistency(self, mocker):
+        # Test that the tiler and tile transform are consistent
+        rng = np.random.default_rng()
+        rnd_tile_size = rng.integers(low=100, high=500)
+        rnd_tile_overlap = rng.random()
+        image_size = rng.integers(low=1000, high=5000)
+        np_image = np.zeros((image_size, image_size, 3), dtype=np.uint8)
+        dm_image = Image.from_numpy(np_image)
+
+        mock_model = MagicMock(spec=Model)
+        mocker.patch("openvino.model_api.tilers.tiler.Tiler.__init__", return_value=None)
+        mocker.patch.multiple(Tiler, __abstractmethods__=set())
+
+        tiler = Tiler(model=mock_model)
+        tiler.tile_size = rnd_tile_size
+        tiler.tiles_overlap = rnd_tile_overlap
+
+        mocker.patch("otx.core.data.dataset.tile.OTXTileTransform.__init__", return_value=None)
+        tile_transform = OTXTileTransform()
+        tile_transform._tile_size = (rnd_tile_size, rnd_tile_size)
+        tile_transform._overlap = (rnd_tile_overlap, rnd_tile_overlap)
+
+        dm_rois = [xywh_to_x1y1x2y2(*roi) for roi in tile_transform._extract_rois(dm_image)]
+        # 0 index in tiler is the full image so we skip it
+        assert np.allclose(dm_rois, tiler._tile(np_image)[1:])
```

Comment:
This seems more suitable for the unit test.

Reply:
I moved this under unit test folder.